### ISSUE TITLE
ClassNotFoundException: Didn't find class "com.chaquo.python.PyException"

### ISSUE THREAD

**Issue by mhsmith** [chaquo/chaquopy#261]
_Originally posted by @tongjinlv in https://github.com/chaquo/chaquopy/issues/43#issuecomment-613360238_

I compiled into release   but is run  error  
error message：
lang.ClassNotFoundException: Didn't find class "com.chaquo.python.PyException" on path: DexPathList[[zip file "/data/app/com.matatalab.matatacode-9_fj_Bw-RN1FBE6tu42f0A==/base.apk"],nativeLibraryDirectories=[/data/app/com.matatalab.matatacode-9_fj_Bw-RN1FBE6tu42f0A==/lib/arm64, /data/app/com.matatalab.matatacode-9_fj_Bw-RN1FBE6tu42f0A==/base.apk!/lib/arm64-v8a, /system/lib64, /product/lib64]]

**Comment by mhsmith** [comment]
Please post the full stack trace, and any relevant source code.

**Comment by tongjinlv** [comment]
The problem is solved due to the release mode compression, thank you very much

**Comment by mhsmith** [comment]
_Originally posted by @zeynabimani in https://github.com/chaquo/chaquopy/issues/358#issuecomment-676397390_

Hi. I have almost the same problem. I just want to start Python in my android application, but when I use this code I get an error.
The code is:

`if (!Python.isStarted()) {
            Python.start(new AndroidPlatform(this));
}`

When I use this code in any part of the program MainActivity or somewhere else I got the error and I don't know why?!

The error is:
```
2020-08-19 18:19:04.135 12263-12263/com.dji.sdk.sample E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.dji.sdk.sample, PID: 12263
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.dji.sdk.sample/com.dji.sdk.sample.internal.controller.MainActivity}: java.lang.ClassNotFoundException: Didn't find class "com.chaquo.python.PyException" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.dji.sdk.sample-ImLlWv7TGO-MqNuoILbVIA==/base.apk", dex file "InMemoryDexFile[cookie=[0, 494810873152]]", dex file "InMemoryDexFile[cookie=[0, 494810874496]]"],nativeLibraryDirectories=[/data/app/com.dji.sdk.sample-ImLlWv7TGO-MqNuoILbVIA==/lib/arm64, /data/app/com.dji.sdk.sample-ImLlWv7TGO-MqNuoILbVIA==/base.apk!/lib/arm64-v8a, /system/lib64]]
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2977)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3114)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:113)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:71)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:201)
        at android.app.ActivityThread.main(ActivityThread.java:6831)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:927)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "com.chaquo.python.PyException" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.dji.sdk.sample-ImLlWv7TGO-MqNuoILbVIA==/base.apk", dex file "InMemoryDexFile[cookie=[0, 494810873152]]", dex file "InMemoryDexFile[cookie=[0, 494810874496]]"],nativeLibraryDirectories=[/data/app/com.dji.sdk.sample-ImLlWv7TGO-MqNuoILbVIA==/lib/arm64, /data/app/com.dji.sdk.sample-ImLlWv7TGO-MqNuoILbVIA==/base.apk!/lib/arm64-v8a, /system/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:171)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at com.chaquo.python.Python.startNative(Native Method)
        at com.chaquo.python.Python.a(:50)
        at com.dji.sdk.sample.internal.controller.MainActivity.onCreate(:113)
        at android.app.Activity.performCreate(Activity.java:7224)
        at android.app.Activity.performCreate(Activity.java:7213)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2957)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3114) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:113) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:71) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:201) 
        at android.app.ActivityThread.main(ActivityThread.java:6831) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:547) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:927) 
```

I read the documentation, and I know one way for starting python is to change the application name in androidManifest.xml file, but I can't do this because I use a Sample Application, and I can't change the application name in the manifest file. 

Can you please help me find the problem?

**Comment by mhsmith** [comment]
> The problem is solved due to the release mode compression, thank you very much

@tongjinlv: can you explain what you mean by this? For example, was it affected by the build variant? What exactly did you do to solve it?

**Comment by mhsmith** [comment]
@zeynabimani: Changing the application name in the AndroidManifest.xml shouldn't be necessary, because they only thing `PyApplication` does is make that same `Python.start` call.

Which devices or emulators have you tried running the app on, and did they all show the same problem?

**Comment by mhsmith** [comment]
OK, I think I've found the cause. Try editing your build.gradle file to change every appearance of `minifyEnabled true` to `minifyEnabled false`.

This will be fixed in the next version of Chaquopy.

**Comment by zeynabimani** [comment]
Thank you very much @mhsmith. 
My problem solved by changing `minifyEnabled  true` to `minifyEnabled  false`.

**Comment by mhsmith** [comment]
This issue is fixed in Chaquopy 9.0.0. To upgrade, edit your app's top-level build.gradle file and change the version number of `com.chaquo.python:gradle`.
